scripts/skills/perks/perk_rf_vanquisher.nut

```diff
@@ -72,22 +72,12 @@ this.perk_rf_vanquisher <- ::inherit("scripts/skills/skill", {
 		}
 	}
 
-	function onAffordablePreview( _skill, _movementTile )
+	function onAfterUpdatePreview( _properties, _previewedSkill, _previewedMovement )
 	{
-		if (this.m.IsInEffect)
-		{
-			foreach (skill in this.getContainer().getAllSkillsOfType(::Const.SkillType.Active))
-			{
-				this.modifyPreviewProperty(skill, "IsSkillUseHalfCost", this.getContainer().getActor().getCurrentProperties().IsSkillUseHalfCost, false);
-			}
-		}
 		// Gain Ground can only be used on a valid tile
-		else if ((_skill != null && _skill.getID() == "actives.rf_gain_ground") || (_movementTile != null && this.isTileValid(_movementTile)))
+		if ((_previewedSkill != null && _previewedSkill.getID() == "actives.rf_gain_ground") || (_previewedMovement != null && this.isTileValid(_previewedMovement.End)))
```

Comment:
The `||` in the middle of this if does not seem correct and should imo be a `&&`. You want all of these conditions to be true in order to change the preview cost, not one or the other.

Reply:
It is correct. The condition is the following:
- The player is previewing a skill and the skill is Gain Ground.
- OR
- The player is previewing a movement and the end tile is valid.

Gain Ground can only be used on a valid tile, so we know for sure that it will be valid in that case.